src/scores/processing/block_bootstrap.py

```diff
@@ -0,0 +1,410 @@
+) -> Tuple[xr.DataArray, ...]:
+    """
+    Repeatedly performs bootstrapping on provided arrays across specified dimensions, stacking
+    the new arrays along a new "iteration" dimension. Bootstrapping is executed in a nested
+    manner: the first provided dimension is bootstrapped, then for each bootstrapped sample
+    along that dimension, the second provided dimension is bootstrapped, and so forth.
+
+    Args:
+        *objects: Data to bootstrap. Multiple datasets can be passed to be bootstrapped
+            in the same way. All input objects must have nested dimensions.
+        blocks: Dictionary of dimension(s) to bootstrap and the block sizes to use
+            along each dimension: `{dim: blocksize}`. Nesting is based on the order of
+            this dictionary.
+        n_iteration: The number of iterations to repeat the bootstrapping process. Determines
+            how many bootstrapped arrays will be generated and stacked along the iteration
+            dimension.
+        exclude_dims: An optional parameter indicating the dimensions to be excluded during
+            bootstrapping for each object provided in `objects`. This parameter expects a list
+            of lists, where each inner list corresponds to the dimensions to be excluded for
+            the respective object. By default, the assumption is that no dimensions are
+            excluded, and all objects are bootstrapped across all specified dimensions in `blocks`.
+        circular: A boolean flag indicating whether circular block bootstrapping should be
+            performed. Circular bootstrapping means that bootstrapping continues from the beginning
+            when the end of the data is reached. By default, this parameter is set to True.
+
+     Returns:
+        Tuple of bootstrapped xarray DataArrays or Datasets, based on the input.
+
+    Note:
+        This function expands out the iteration dimension inside a universal function.
+        However, this may generate very large chunks (multiplying chunk size by the number
+        of iterations), causing issues for larger iterations. It's advisable to apply this
+        function in blocks using 'block_bootstrap'.
+
+    References:
+    Wilks, Daniel S. Statistical methods in the atmospheric sciences. Vol. 100.
+      Academic press, 2011.
+    """
+
+    objects_list = list(objects)
+
+    # Rename exclude_dims so they are not bootstrapped
+    if exclude_dims is None:
+        exclude_dims = [[] for _ in range(len(objects_list))]
+    if (
+        not isinstance(exclude_dims, list)
+        or len(exclude_dims) != len(objects_list)
+        or not all(isinstance(x, list) for x in exclude_dims)
+    ):
+        raise ValueError(
+            "exclude_dims should be a list of the same length as the number of "
+            "objects containing lists of dimensions to exclude for each object"
+        )
+    renames = []
+    for i, (obj, exclude) in enumerate(zip(objects_list, exclude_dims)):
```

Comment:
What should be the expected behaviour if a dimension to exclude is also named `"dim{x}"` and `x == ii`?

Is there a reason for renaming as opposed to dropping/splitting and re-combining?